internal/resources/grafana/resource_folder.go

```diff
@@ -138,7 +145,29 @@ func ReadFolder(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 func DeleteFolder(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*common.Client).GrafanaAPI
 
-	if err := client.DeleteFolder(d.Get("uid").(string)); err != nil {
+	deleteParams := []url.Values{}
+	if d.Get("prevent_destroy_if_not_empty").(bool) {
+		// Search for dashboards and fail if any are found
+		dashboards, err := client.FolderDashboardSearch(url.Values{
+			"type":      []string{"dash-db"},
+			"folderIds": []string{d.Id()},
+		})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if len(dashboards) > 0 {
+			var dashboardNames []string
+			for _, dashboard := range dashboards {
+				dashboardNames = append(dashboardNames, dashboard.Title)
+			}
+			return diag.Errorf("folder %s is not empty and prevent_destroy_if_not_empty is set. It contains the following dashboards: %v", d.Get("uid").(string), dashboardNames)
+		}
+	} else {
+		// If we're not preventing destroys, then we can force delete folders that have alert rules
+		deleteParams = append(deleteParams, gapi.ForceDeleteFolderRules())
+	}
```

Comment:
Can we omit the `else`?
```suggestion
	}

	// If we're not preventing destroys, then we can force delete folders that have alert rules
	deleteParams = append(deleteParams, gapi.ForceDeleteFolderRules())
```

Reply:
We cannot. The param should only be there if `prevent_destroy_if_not_empty` is false

Reply:
Oohh right, I missed the lack of `return` in the block above 🤦🏽